### PR TITLE
HDDS-10445. OM admin CLI to enable force fetching network topology tree from SCM

### PR DESCRIPTION
**Pull request by tanvipenumudy**
## What changes were proposed in this pull request?

OM has a background thread running that periodically refetches the updated network topology layout information from SCM for the refresh duration frequency of every one hour (default).

This change introduces an OM admin CLI for force fetching the network topology tree information from SCM (on demand) without having to rely on the configuration: `ozone.om.network.topology.refresh.duration`.

CLI usage:
```
ozone admin om fetch-topology-tree -id=omServiceId
```

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10445

## How was this patch tested?

- Added integration test.
- Green Git CI: https://github.com/tanvipenumudy/ozone/actions/runs/8108558196

### CODE REVIEW

**Comment by tanvipenumudy** [comment]
@kerneltime, @duongkame could you please take a look? Thanks!

**Comment by ArafatKhan2198** [comment]
Thanks for working on this, @tanvipenumudy. I don't see any tests written for it yet. Would you mind adding an integration test to verify your results, please?

**Comment by tanvipenumudy** [comment]
@adoroszlai could you please review the changes as well? Thanks!

File: hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto [hunk 1]
```diff
@@ -147,6 +147,7 @@ enum Type {
   ListStatusLight = 129;
   GetSnapshotInfo = 130;
   RenameSnapshot = 131;
+  RefetchNetworkTopologyTree = 132;
```
Comment (adoroszlai) [inline hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto:150]: Let's omit `Tree` to make it a bit shorter.  (Also in request/response, method and command names.)

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/FetchNetworkTopologyTreeSubCommand.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        status = client.refetchNetworkTopologyTree();
+      } catch (IOException e) {
+        System.err.println("Force fetching network topology tree information " +
+            "has failed: " + e.getMessage());
+      }
+      if (status) {
+        System.out.println(
+            "Force fetching network topology tree information " +
+                "is complete.");
+      }
```
Comment (adoroszlai) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/om/FetchNetworkTopologyTreeSubCommand.java:61]: I find this output a bit too verbose.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java [hunk 3]
```diff
@@ -1094,11 +1094,17 @@ private void stopSecretManager() {
...
 
+  @Override
   public UUID refetchSecretKey() {
     secretKeyClient.refetchSecretKey();
     return secretKeyClient.getCurrentSecretKey().getId();
   }
 
+  @Override
+  public boolean refetchNetworkTopologyTree() {
+    return scmTopologyClient.refetchClusterTree(configuration);
```
Comment (adoroszlai) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:1105]: I think admin privilege should be required to make the request.

**Review by adoroszlai — COMMENTED** [review]
Thanks @tanvipenumudy for working on this.

The new command seems to work only with OM HA.  I don't think this functionality should be limited to such setup.

**Comment by adoroszlai** [comment]
@tanvipenumudy can we close this?

**Comment by tanvipenumudy** [comment]
Sure, marking the PR as closed.

**Pull request closed without merging**